api/proto/teleport/plugins/v1/plugin_service.proto

```diff
@@ -138,6 +138,26 @@ message SearchPluginStaticCredentialsResponse {
   repeated types.PluginStaticCredentialsV1 credentials = 1;
 }
 
+// NeedsCleanupRequest is the request type for NeedsCleanup.
+message NeedsCleanupRequest {
+  // Type is the plugin type. We only need the string representation of the plugin type and not
+  // the PluginType message, as we don't want the oauth client ID here.
+  string type = 1;
```

Comment:
should types like this be represented as an enum?

Reply:
Perhaps, but unfortunately I think that ship has sailed -- other instances where we've got a plugin type are represented as strings.